### ISSUE TITLE
ActionView::Helpers::AssetTagHelper javascript/css link helpers break if the request object is nil

### ISSUE THREAD

**Issue by tcannonfodder** [rails/rails#31265]
### Steps to reproduce

1. Create a default mailer layout (`mailer.html.erb`)
2. Use `stylesheet_link_tag` in the layout (e.g:`<%= stylesheet_link_tag "mailer", media: "all" %>`)
3. Attempt to generate an email
4. Observe that the following exception is raised:
```
ActionView::Template::Error: undefined method `send_early_hints' for nil:NilClass
```

### Expected behavior
The link should be generated without an exception, and without early hints.

### Actual behavior
The helper attempts to call `request.send_early_hints`, which raises an exception because `request` is `nil`: https://github.com/rails/rails/blob/v5.2.0.beta2/actionview/lib/action_view/helpers/asset_tag_helper.rb#L143

### System configuration
**Rails version**: 5.2.0.beta2

**Ruby version**: Ruby 2.4.0

**Comment by mokolabs** [comment]
I just ran into this issue while setting up a mailer in a new app using Rails 5.2beta2.

Why was this issue closed without comment? Is there a fix or workaround in place?

Thanks!

**Comment by madrobby** [comment]
@mokolabs See above, it was fixed and committed in 6121463 (idk if it made it into a beta yet)
